Source/WebKit/UIProcess/UserContent/WebUserContentControllerProxy.cpp

```diff
@@ -215,19 +219,45 @@ void WebUserContentControllerProxy::removeAllUserScripts(API::ContentWorld& worl
     });
 }
 
+#if ENABLE(WK_WEB_EXTENSIONS)
+void WebUserContentControllerProxy::removeAllUserScripts(RemoveWebExtensions removeWebExtensions)
+#else
 void WebUserContentControllerProxy::removeAllUserScripts()
+#endif
 {
-    HashCountedSet<RefPtr<API::ContentWorld>> worlds;
-    for (auto userScript : m_userScripts->elementsOfType<API::UserScript>())
-        worlds.add(const_cast<API::ContentWorld*>(&userScript->contentWorld()));
+#if ENABLE(WK_WEB_EXTENSIONS)
+    if (removeWebExtensions == RemoveWebExtensions::Yes)
+#endif
+    {
+        HashCountedSet<RefPtr<API::ContentWorld>> worlds;
+        for (auto userScript : m_userScripts->elementsOfType<API::UserScript>())
+            worlds.add(const_cast<API::ContentWorld*>(&userScript->contentWorld()));
 
-    auto worldIdentifiers = WTF::map(worlds, [](auto& entry) {
-        return entry.key->identifier();
-    });
-    for (auto& process : m_processes)
-        process.send(Messages::WebUserContentController::RemoveAllUserScripts(worldIdentifiers), identifier());
+        auto worldIdentifiers = WTF::map(worlds, [](auto& entry) {
+            return entry.key->identifier();
+        });
+
+        for (auto& process : m_processes)
+            process.send(Messages::WebUserContentController::RemoveAllUserScripts(worldIdentifiers), identifier());
+
+        m_userScripts->elements().clear();
+
+        return;
+    }
+
+#if ENABLE(WK_WEB_EXTENSIONS)
+    ASSERT(removeWebExtensions == RemoveWebExtensions::No);
+
+    Vector<Ref<API::UserScript>> scriptsToRemove;
+    for (auto userScript : m_userScripts->elementsOfType<API::UserScript>()) {
+        if (removeWebExtensions == RemoveWebExtensions::No && WebExtensionMatchPattern::isWebExtensionURL(userScript->userScript().url()))
```

Comment:
Do we typically check for something we asserted to be true above? If not, then you could remove this check here and the other places below. If so, then you should add this check in `WebUserContentControllerProxy::removeAllContentRuleLists()` as well.

Reply:
The assert was for the opposite case, to make sure more enum values don't get added. I'll add it to the other place.